feat(card): use core tokens

mlogsdon18 opened the pull request:
## Description

This PR updates the Card component to utilize the new core tokens.

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Expected Design Changes
- Smaller card no longer exists (card minimum width has been updated to allow for smaller cards)
- Quiet card padding has been removed
- Card title is smaller
- Divider added underneath image rather than a border

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply.  If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

mlogsdon18 commented:
@Westbrook this is ready for your review! I believe that all of the design changes flagged in the VRTs are expected, which I listed out in the PR description.

Westbrook commented:
https://card-core-tokens--spectrum-web-components.netlify.app/storybook/?path=/story/card--quiet-file and https://pr-1693--spectrum-css.netlify.app/card.html#quietfile don't feel like they are giving the same output here. Initial point of research would be that there's not `display` value on the `sp-card`, the CSS demo uses a `div` which gets `display: block` by default. Beyond that, there still feels like some other places where the two versions lack alignment.

mlogsdon18 commented:
@najikahalsema tagging you in here for review if you get a chance! Basically Westbrook had some concerns but I'm fairly certain they are all intended design changes. Could you take a look and confirm that for me or see if there are changes we should make to the examples to account for this?

najikahalsema commented:
> @najikahalsema tagging you in here for review if you get a chance! Basically Westbrook had some concerns but I'm fairly certain they are all intended design changes. Could you take a look and confirm that for me or see if there are changes we should make to the examples to account for this?

I did check that removing the set width did narrow it to 100px. I wonder why they changed the width to be 100 px when it seems like all the examples superimpose 200px or greater anyway. 

My other question is whether the [border seen around the image in the quiet variant](https://pr-1693--spectrum-css.netlify.app/card.html#quiet) is still there (or can be modified to be there in the story). [Here's](https://82379110a792d2d4b40c32477a466227--spectrum-web-components.netlify.app/review/#CardStories/Quiet.png) the diff I'm talking about. This is the biggest change to me and I think it would help to see that in the story so that the VRTs are more accurate.

Westbrook commented:
For my note, it seems I missed that the CSS page puts `width` directly on the Card, were as we were previously applying `width` to the parent: https://github.com/adobe/spectrum-web-components/blob/a5958eb74e6ea2100e02d35b088b280a2fb13cc4/packages/card/stories/card.stories.ts#L158-L159 If we move this inline CSS down to the `sp-card`, _maybe_ it normalizes the two deliveries?

mlogsdon18 commented:
@najikahalsema @Westbrook So the padding around the image for the Quiet variant was removed. The new design is for the preview image to just show based on the aspect ratio of the image. I did find `object-fit: cover` being applied to the image with some custom CSS so I adjusted that to only apply if the Card is not quiet. Now you can see some of the background behind it [in the VRTs](https://0666f0052affd89142edbf6fcde31d3b--spectrum-web-components.netlify.app/review/#CardStories/Quiet.png). 

I also updated the stories to apply the width to the Card itself like you suggested, so they are a lot closer now!

mlogsdon18 commented:
> Something about quiet (https://55ae78d940a8f2f1d56c25b3c468361e--spectrum-web-components.netlify.app/review/#CardStories/Quiet.png) does not align with the CSS implementation (https://pr-1693--spectrum-css.netlify.app/card.html#quiet).

@Westbrook Yes so this was what I was talking about in my comment earlier. The only reason the CSS image is not showing at full size is because it has inline styles set which you can see if you expand the code on that example:
```<img class="spectrum-Asset-image" src="img/example-ava.jpg" style="max-width: 75%; max-height: 75%; object-fit: contain;">```
The design for quiet was updated to no longer include padding and just display the image at full size. The example on the Spectrum CSS docs site is a square image and is set to only display at 75%. The SWC image is not a square image and is currently set to display at 100%. I can add additional inline styles to the SWC example if you'd like to get it closer but that is why they're different.

mlogsdon18 commented on packages/card/src/Card.ts:
```diff
@@ -228,6 +234,11 @@ export class Card extends LikeAnchor(
             </sp-asset>
+            ${this.variant !== 'quiet' && !this.horizontal
+                ? html`
+                      <sp-divider size="s"></sp-divider>
```
From the design, I was under the impression that any image should have a divider underneath it. Are you saying here that the gallery variant should not?
![Screenshot 2023-06-27 at 9 34 48 AM](https://github.com/adobe/spectrum-web-components/assets/99203545/899726c5-fa92-4ba8-8e29-a1c86f0ff8a5)

mlogsdon18 replied:
Our examples in Spectrum CSS didn't distinguish `coverPhoto` vs `preview` very well so I assumed they both needed it, but are there different use cases?

Westbrook replied:
I don't know how to answer that question. It is not present in your [PR preview](https://pr-1693--spectrum-css.netlify.app/card-gallery.html) for CSS, which is why I asked.

mlogsdon18 replied:
Oh good callout. I must have just forgotten to add it to that example since it's separate from the others. According to the design, all cards except for quiet ones should have a divider underneath the image. I'll make that update to my PR to add it into that example.

Westbrook commented:
Thanks for clarifying (multiple times 🙈) the quiet updates.

If you could do the same on the Gallery divider, we should be good to go here.

Westbrook commented on packages/card/src/spectrum-card.css:
```diff
@@ -11,231 +11,238 @@ governing permissions and limitations under the License.
+    position: absolute;
+}
+:host:after {
+    block-size: 100%;
+    border: 0 solid #0000;
+    border-radius: var(
+        --mod-card-corner-radius,
+        var(--spectrum-card-corner-radius)
     );
-    --spectrum-card-coverphoto-height: var(
-        --spectrum-card-m-coverphoto-height,
-        var(--spectrum-global-dimension-size-1700)
+    content: '';
+    inline-size: 100%;
+    inset-block-start: 0;
+    inset-inline: 0;
+    margin-block-start: calc(
+        var(
+                --mod-card-focus-indicator-width,
+                var(--spectrum-card-focus-indicator-width)
+            ) * -1
     );
-    --spectrum-card-coverphoto-border-bottom-size: var(
-        --spectrum-card-m-coverphoto-border-bottom-size,
-        var(--spectrum-alias-border-size-thin)
+    margin-inline-start: calc(
+        var(
+                --mod-card-focus-indicator-width,
+                var(--spectrum-card-focus-indicator-width)
+            ) * -1
     );
-    --spectrum-card-checkbox-margin: var(
-        --spectrum-card-m-checkbox-margin,
-        var(--spectrum-global-dimension-size-200)
+    position: absolute;
+}
```
@mlogsdon18 sorry I missed this last one, but these pseudo elements need to have `pointer-events: none;` or they block interaction with other parts of the page. Do you want to work out a new version of this package now, or should we add this as an override locally in anticipation for that change coming along later?

Westbrook replied:
For reference, this is what was causing [these](https://app.circleci.com/pipelines/github/adobe/spectrum-web-components/16382/workflows/83434957-27fd-4a1a-b69d-f354113b673d/jobs/350359/tests) sorts of unit test failures.

Westbrook replied:
I fixed this locally, but we'll need this updated in the CSS source, sooner or later. It actually aligns with [this issue](https://github.com/adobe/spectrum-web-components/issues/3357) in Dropzone.

Westbrook reviewed (approved):
LGTM! Thanks for this @mlogsdon18. Let us know if you need any support in getting those `pointer-events: none` changes added upstream for a future PR. 🙇🏼
